tensorflow/lite/kernels/fully_connected.cc

```diff
@@ -64,6 +64,48 @@ constexpr int kBiasTensor = 2;
 constexpr int kOutputTensor = 0;
 constexpr int kShuffledInputWorkspaceTensor = 1;
 
+inline bool CheckDtype(const TfLiteTensor* input, const TfLiteTensor* filter,
+                       const TfLiteTensor* bias, TfLiteTensor* output,
+                       TfLiteFullyConnectedParams* params) {
+  const bool is_Quantized =
+      ((filter->type == kTfLiteUInt8) || (filter->type == kTfLiteInt8));
+  const bool is_Hybrid = is_Quantized && (input->type == kTfLiteFloat32);
+  const bool is_Shuffled =
+      is_Quantized && (params->weights_format ==
+                       kTfLiteFullyConnectedWeightsFormatShuffled4x16Int8);
+  const bool is_BiasNotFloatType = bias && (bias->type != kTfLiteFloat32);
+  const bool is_BiasNotIntType = bias && (bias->type != kTfLiteInt32);
+
+  if (is_Quantized) {
```

Comment:
Can we make this function return a status and use TFLITE_ENSURE_EQ for all of these checks so that we get more meaningful error info?